doc/source/glossary.rst

```diff
@@ -166,9 +166,10 @@ Glossary
 
 
    array scalar
-       For uniformity in handling operands, NumPy treats
-       a :doc:`scalar <reference/arrays.scalars>` as an array of zero
-       dimension.
+       An :doc:` array scalar <reference/arrays.scalars>` is an instance of the types/classes float32, float64, 
+       etc.. For uniformity in handling operands, NumPy treats a scalar as 
+       an array of zero dimension. In contrast, a 0-dimensional array is an :doc:` ndarray <reference/arrays.ndarray>` instance 
+       containing precisely one array scalar. 
```

Comment:
The extended description is great, until _"containing precisely one array scalar"_ which seems imprecise. An ndarray doesn't contain an array scalar - there's no instance of a dtype type inside the array container, just a number. How about rephrasing that to _"containing precisely one value"_?